agorauikit_android/src/main/java/io/agora/agorauikit_android/AgoraVideoViewer.kt

```diff
@@ -228,6 +267,345 @@ open class AgoraVideoViewer: FrameLayout {
+      "uikit": {
+        "platform": "Android",
+        "framework": "Kotlin",
+        "version": "1.0.0"
+      }
+    }"""
+        var message: RtmMessage = agRtmClient.createMessage()
+        message.text = userData
+
+
+        val option = SendMessageOptions()
+        option.enableOfflineMessaging = true
+
+        if (!toChannel) {
+            agRtmClient.sendMessageToPeer(
+                peerRtmId,
+                message,
+                option,
+                object : ResultCallback<Void> {
+                    override fun onSuccess(p0: Void?) {
+                        println("UserData message sent to $peerRtmId")
+                    }
+
+                    override fun onFailure(p0: ErrorInfo?) {
+                        println("Failed to send UserData to peer: $peerRtmId")
+                    }
+                })
+        } else {
+            agRtmChannel.sendMessage(message, option, object : ResultCallback<Void> {
+                override fun onSuccess(p0: Void?) {
+                    println("USerData message sent to channel")
+                }
+
+                override fun onFailure(p0: ErrorInfo?) {
+                    println("Failed to send UserData to channel")
+                }
+            })
+        }
+
+    }
+
+    internal fun askForUserMic(peerRtcId: Int, isMicEnabled: Boolean) {
+        var peerRtmId: String? = null
+        var json: String = """{
+        "messageType": "MuteRequest",
+        "rtcId": $peerRtcId,
+        "mute": $isMicEnabled,
+        "device": "0",
+        "isForceFul": "false"
+    }"""
+
+        var message: RtmMessage = agRtmClient.createMessage()
+        message.text = json
+
+        val option = SendMessageOptions()
+        option.enableOfflineMessaging = true
+
+        if (peerRtcId == userID) {
+            println("Can't send message to the local user")
+        } else {
+            if (this.agoraSettings.uidToUserIdMap.containsKey(peerRtcId)) {
+                peerRtmId = this.agoraSettings.uidToUserIdMap.getValue(peerRtcId)
+
+                agRtmClient.sendMessageToPeer(
+                    peerRtmId,
+                    message,
+                    option,
+                    object : ResultCallback<Void> {
+                        override fun onSuccess(p0: Void?) {
+                            println("Mic Request message sent to $peerRtmId")
+                        }
+                        override fun onFailure(p0: ErrorInfo?) {
+                            println("Failed to send mic request message to $peerRtmId. Error : $p0")
+                        }
+                    })
+            }
+        }
+    }
+
+    internal fun askForUserCamera(peerRtcId: Int, isCameraEnabled: Boolean) {
+        var peerRtmId: String? = null
+        var json: String = """{
+        "messageType": "CameraRequest",
+        "rtcId": $peerRtcId,
+        "mute": $isCameraEnabled,
+        "device": "1",
+        "isForceFul": "false"
+    }"""
```

Comment:
This should be generated from a class - less chance of mistakes if one of these gets updated later